### ISSUE TITLE
Filtering store updates / using redux middleware

### ISSUE THREAD

**Issue by SebT** [Akryum/vue-apollo#118]
Hi.

**TL;DR**: How one can filter apollo's store updates ?

**The problem**
I use web socket to subscribe to some update events. Let's say I have an Article apollo type with an id and a content.

So I use `$apollo.subscribe` to subscribe to event with websockets. That way, my store automatically updates the store when an event is received. 

The problem is that events are not necessarily in the right order, and there is no way I can enforce events to arrive in the same order they were emitted. Thus sometimes, the article can be unsynced for users.

**Possible solution**
I use mongodb to store my articles. So I have a `__v` property that is incremented on each update. I'd like to use a kind of hook to prevent the store to update the article with id 'x' if the `__v'`field from the event is lesser the article's `__v` field that is already in my apollo store.

I think the simpliest way to implement  this is to register a custom middleware on apollo's redux store that prevents store update if a more recent event already updated the store.

How can I achieve that ?
